### DISCUSSION TITLE
SSl handshake error when client connect to kafka cluster

### DISCUSSION THREAD

**Discussion by bestvarut** [strimzi/strimzi-kafka-operator#6357] [Q&A] [answered]
I have SSL handshake problem to connect port 9093 which have tls-encryption enable
[cluster.zip](https://github.com/strimzi/strimzi-kafka-operator/files/8056291/cluster.zip)
YAML for `Kubectl apply -f cluster.yaml`
as 9093 type is Nodeport so I already add
`ssl.endpoint.identification.algorithm: ''` to disable hostname verification
but it still doesnt work when client cennect to kafka cluster port 9093
![image](https://user-images.githubusercontent.com/51030057/153768548-a1393d48-1507-47e0-9f2f-c24d20bfd4b2.png)
here is my client-ssl.properties
```
bootstrap.servers=10.233.8.79:9093
security.protocol=SSL
ssl.truststore.location=/home/varutph/test_ssl/kafka.client.truststore.jks
ssl.truststore.password=clientpass
```
I have use create my own CA way from https://github.com/scholzj/strimzi-custom-ca-test

### How i create truststore
-After I complete above instruction and deploy my kafka cluster
-I extract ca.crt from cluster-ca-cert secret
![image](https://user-images.githubusercontent.com/51030057/153768745-ed911e43-f019-4bb7-bd52-379feb4c3d71.png)

```
kubectl get secret best-kafka-cluster-cluster-ca-cert -o jsonpath='{.data.ca\.crt}' | base64 --decode > ca.crt
keytool -keystore kafka.client.truststore.jks -alias ca.crt -import -file ca.crt  -storepass clientpass -keypass clientpass
```

### also in logs of cluster-operator show this error
![image](https://user-images.githubusercontent.com/51030057/153767941-cb958ad2-0eb5-45d0-a294-883eedc3260a.png)

like it ignore that config `ssl.endpoint.identification.algorithm: ''`

### **Please help me on this issue**

**Comment by scholzj** [comment, marked as answer]
I cannot unpack the ZIP file. Maybe you can paste the YAMLs and logs as code instead of using ZIPs and pictures. It is much easier and secure to work with. 

I'm not entirely sure from your description what are you doing. But:
* If you are connecting to port 9093, you are not connecting to nodeport => node port would be something like `32586`
* The `ssl.endpoint.identification.algorithm` setting needs to be set in your client when using node ports. Not in the brokers.
